### ISSUE TITLE
chore: do we need to use slug.Make(opt.Name)

### ISSUE THREAD

**Issue by meling** [quickfeed/quickfeed#1107]
We currently only use `slug.Make(opt.Name)` when looking up an organization (from the `scm`); at this point it might already be too late, since the organization has been created on GitHub and thus, whatever the user enters when looking up the course via our frontend, it would already have been sluggified and the user should probably be aware... 

Previously we also used `slug.Make` when looking up a repository, but this was not used when creating repositories and there could an argument made for doing the slugging before adding the repository name to the database as well.

@JosteinLindhom thoughts?

**Comment by meling** [comment]
We do have similar functionality implemented for limiting what students can do for group names. The question is whether or not we could replace our group name logic with `slug.Make()` with lower case disabled.

**Comment by JoachimTislov** [comment]
To my understanding does slug.Make convert norwegian charaters into english ones (æ -> ae, ø -> o, å -> aa). So doing this for repository and group names would be reasonable, but I don't see the issue and reasoning behind converting norwegian to english characters, does it cause issues? I don't think a string would affect anything... im not sure if I understand the problem here

Group name solution, not with this: `lower case disabled.`
file: scm_helpers.go:36
```go 
func createRepo(ctx context.Context, sc scm.SCM, course *qf.Course, group *qf.Group) (*qf.Repository, error) {
        groupName := slug.Make(group.GetName())
	opt := &scm.GroupOptions{
		Organization: course.ScmOrganizationName,
		GroupName:    groupName ,
		Users:        group.UserNames(),
	}
```

Repository name solution:
github.go:380
```go
repositoryName := slug.Make(opt.Repo)
repo, _, err = s.client.Repositories.Create(ctx, opt.Owner, &github.Repository{
	Name:    github.String(repositoryName),
	Private: github.Bool(opt.Private),
})
```

**Comment by meling** [comment]
We have two uses of `slug.Make`:

1. In `scm/github_mock_opts.go`:
```go
var toOrg = func(course *qf.Course) github.Organization {
	return github.Organization{
		ID:    github.Int64(int64(course.ScmOrganizationID)),
		Login: github.String(slug.Make(course.ScmOrganizationName)),
	}
}
```
2. In `scm/github.go`:
```go
func (s *GithubSCM) GetOrganization(ctx context.Context, opt *OrganizationOptions) (*qf.Organization, error) {
...
	} else {
		name := slug.Make(opt.Name)
		githubOrg, _, err = s.client.Organizations.Get(ctx, name)
```
I will argue that we need neither of these.
1. Is obvious, since all test examples are already sluggified.
2. See the [first comment above](https://github.com/quickfeed/quickfeed/issues/1107#issue-2372566062). That is, there is no need to do it for the course organization, since by the time it reaches the backend, the organization has already been created on GitHub.

This issue was originally about the possible removal of a dependency (since we only use the `slug.Make` function once in real code).

One could argue that we could use a slug thing in the frontend when students are creating a group name. We do have some logic for that in `web/groups.go`:
```go
// checkGroupName returns an error if the group name is invalid; otherwise nil is returned.
func (s *QuickFeedService) checkGroupName(courseID uint64, groupName string) error {
```
There is also a test in `web/groups_name_test.go`:
```go
func TestBadGroupNames(t *testing.T) {
```
The frontend implements what appears to be the same logic in `public/src/Helpers.ts`:
```ts
export const validateGroup = (group: CourseGroup): { valid: boolean, message: string } => {
```
And there is a test:
```ts
describe('validateGroup', () => {
```

If these checks are not enough to match the requirements of a GitHub repository name, then we should fix our own checks.

**In conclusion, let's remove the slug dependency, and if needed augment our existing group name checks.**

**Comment by JoachimTislov** [comment]
> This issue was originally about the possible removal of a dependency (since we only use the slug.Make function once in real code).

Understood, sorry for the misunderstanding earlier, refer to #1228
